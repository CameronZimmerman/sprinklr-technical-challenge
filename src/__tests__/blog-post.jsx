import React from "react";
import TestRenderer from "react-test-renderer";
import Blog from "../templates/blog.js";

//our test is unable to access the page context where our query lives, so we will manually pass this down as props
const fakePageData = {
  contentfulBlogPost: {
    title: "Test 1 Contentful",
    publishedDate: "2021-09-01",
    body: {
      raw: `{"nodeType":"document","data":{},"content":[{"nodeType":"paragraph","content":[{"nodeType":"text","value":"Test Contentful Post!","marks":[],"data":{}}],"data":{}},{"nodeType":"heading-2","content":[{"nodeType":"text","value":"With Headings!","marks":[],"data":{}}],"data":{}},{"nodeType":"paragraph","content":[{"nodeType":"text","value":"","marks":[],"data":{}}],"data":{}},{"nodeType":"paragraph","content":[{"nodeType":"text","value":"i'm so excited. ","marks":[{"type":"italic"}],"data":{}}],"data":{}},{"nodeType":"paragraph","content":[{"nodeType":"text","value":"","marks":[],"data":{}}],"data":{}}]}`,
    },
  },
  markdownRemark: {
    fields: {
      slug: "MarkdownTest1",
    },
    frontmatter: {
      title: "Test 1 Markdown",
      publishedDate: "2021-09-01",
    },
    html: "<p>Test</p>",
  },
};

describe("Blog Page", () => {
  it("renders blog post correctly", () => {
    const testRenderer = TestRenderer.create(<Blog data={fakePageData} />);
    const blogPostTestInstance = testRenderer.toJSON();
    expect(blogPostTestInstance).toMatchSnapshot();
  });
});
