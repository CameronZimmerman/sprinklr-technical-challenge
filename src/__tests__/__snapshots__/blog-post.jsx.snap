// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Page renders blog post correctly 1`] = `
<div>
  <section>
    <header>
      <a
        href="/"
      >
        <h1>
          Test Title
        </h1>
      </a>
      <nav>
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <h1>
      Test 1 Contentful
    </h1>
    <p>
      2021-09-01
    </p>
    <p>
      Test Contentful Post!
    </p>
    <h2>
      With Headings!
    </h2>
    <p>
      
    </p>
    <p>
      <i>
        i'm so excited. 
      </i>
    </p>
    <p>
      
    </p>
  </section>
  <footer>
    <p>
      Created by 
      Test Author
      , Â© 2021
    </p>
  </footer>
</div>
`;
